examples/embassy-net/build.rs

```diff
@@ -0,0 +1,17 @@
+fn main() {
+    if cfg!(target_os = "linux") {
+        let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+
+        // Add back the default linker script.
+        println!("cargo:rustc-link-search=/usr/lib/{target_arch}-linux-gnu/ldscripts");
+        println!("cargo:rustc-link-arg=-Telf_{target_arch}.x");
```

Comment:
Probably, `elf_x86_64.x` should be added for this example?

Reply:
As in including a GNU linker script in the sntpc repo? I'm kind of against that as it'd require us to pin the target and there's folks developing on Mx Mac machines out there.

Reply:
My understanding is that `elf_<target_arch>.x` is not a standard one. E.g. on Fedora 41 I use now, there is no such a package that provides `/usr/lib/x86-64-linux-gnu/ldscripts`. So, it is worth providing a sample file. If someone has a different platform, they will be able to overwrite the default linker script with another one by specifying a respective file.

Reply:
I've added the binutils linker scripts for x86_64 and aarch64, and they can easily be overridden so I think this should hopefully do the the trick.